src/App.tsx

```diff
@@ -11,27 +12,44 @@ interface itemsTypes {
   isEmpty: boolean;
 }
 
+const initialItems = [
+  {
+    todoId: 1,
+    content: "Item 1",
+    isChecked: false,
+    isEmpty: false,
+  },
+  {
+    todoId: 2,
+    content: "Item 2",
+    isChecked: false,
+    isEmpty: false,
+  },
+  {
+    todoId: 3,
+    content: "Item 3",
+    isChecked: false,
+    isEmpty: false,
+  },
+];
+
 const App = () => {
-  const [items, setItems] = useState<itemsTypes[]>([
-    {
-      todoId: 1,
-      content: "Item 1",
-      isChecked: false,
-      isEmpty: false,
-    },
-    {
-      todoId: 2,
-      content: "Item 2",
-      isChecked: false,
-      isEmpty: false,
-    },
-    {
-      todoId: 3,
-      content: "Item 3",
-      isChecked: false,
-      isEmpty: false,
-    },
-  ]);
+  const [items, setItems] = useState<itemsTypes[]>(initialItems || []);
+
+  useEffect(() => {
+    const itemsFromLocalStorage = localStorage.getItem("items");
+    const jsonData = JSON.parse(itemsFromLocalStorage || "");
+
+    if (itemsFromLocalStorage) {
+      setItems(jsonData);
+    }
+  }, []);
+
+  useEffect(() => {
+    setTimeout(() => {
+      localStorage.setItem("items", JSON.stringify(items));
+    }, 2000);
+  }, [items]);
```

Comment:
This code will set the local storage items every time the `items` state change. It is very expensive when you imagine that this would be an API call.
Adding the 
```suggestion
  useEffect(() => {
    const debouncingDelay = setTimeout(() => {
      localStorage.setItem("items", JSON.stringify(items));
    }, 2000);
    return ()=>clearTimeout(debouncingDelay)
  }, [items]);
```